src/mmrelay/db_utils.py

```
@@ -771,24 +783,117 @@ def _store(cursor: sqlite3.Cursor) -> None:
+        table (str): Table name ('longnames' or 'shortnames').
+        current_ids (set[str]): Set of Meshtastic node IDs that are currently valid.
+
+    Returns:
+        int: Number of rows deleted.
+    """
+    if not current_ids:
+        return 0
+
+    placeholders = ",".join("?" * len(current_ids))
+    cursor.execute(
+        f"DELETE FROM {table} WHERE meshtastic_id NOT IN ({placeholders})",
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)